### ISSUE TITLE
eventDrop, access old resource

### ISSUE THREAD

**Issue by arshaw** [fullcalendar/fullcalendar-scheduler#39]
from @danielsmacdonald, moved from [here](https://github.com/fullcalendar/fullcalendar/issues/2830)

> The eventDrop callback has these arguments: (event, delta, revertFunc, jsEvent, ui, view). When used with Scheduler, event.resourceId represents the new location of the event. Could we have something to represent the old location (resourceId before the move)?
> 
> Maybe replace delta with prevEvent object? Or just add a property to event, something like event.prevResourceId or event.prevLocation.
> 
> Currently I'm adding the resourceId to the event in duplicate (one with a different name) so only one gets overridden and I can still read the old one. But this is messy and error prone.

**Comment by arshaw** [comment]
from @sboele:

> However, that only mentions eventDrop. I think it would be good to have this information on dragStart and eventAllow also. Probably on all the drag/drop events this would be usefull information,

**Comment by sboele** [comment]
I have found a work around to access the resource where the drag event was started from.

```
eventDragStart: function (event, jsEvent, ui, view) {
	// user started dragging, searching for possible timeslots
	view.prepareHits();
	var hit = view.queryHit(jsEvent.pageX, jsEvent.pageY);
	var cell = view.getHitSpan(hit);

	var clickedResource = cell.resourceId;
}
```

We use the information about the resource where the drag was started from to determine the possible dropping positions.

**Comment by wackawacka** [comment]
sboele - view.getHitSpan is undefined for me - but I was able to use:-

		eventDragStartObj = event;
		view.prepareHits();
		var hit = view.queryHit(jsEvent.pageX, jsEvent.pageY);
		eventDragStartObj.clickedResourceId = view.getHitEl(hit).data('resourceId');

(eventDragStartObj is just a global variable)

**Comment by pierresh** [comment]
That feature is really needed for events with several resources... Now it is painful to know which resource has been removed from the event, and which one has been added ... Thanks! ^^

**Comment by arshaw** [comment]
implemented in the v4 branch. [eventDrop](https://fullcalendar.io/docs/v4/eventDrop) has the properties `event` and `prevEvent` (which will be renamed `oldEvent`) for which you can inspect their resources via `.getResources()`

also implemented are `oldResource` and `newResource` properties which are populated only when the resource is changed.

I'll comment on this ticket again when v4 is officially released.

**Comment by arshaw** [comment]
this has been released in the next alpha, [v4.0.0-alpha.4](https://github.com/fullcalendar/fullcalendar-scheduler/releases/tag/v4.0.0-alpha.4)

**Comment by arshaw** [comment]
[V4 was officially released a few days ago!](https://fullcalendar.io/blog/2019/03/v4-officially-released) This is now officially possible.
